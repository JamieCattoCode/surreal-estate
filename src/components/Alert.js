import React from 'react';
import PropTypes from 'prop-types';
import '../styles/alert.css';

const Alert = ({ message, success }) => {
  let colour;
  if (success) {
    colour = 'green';
  } else {
    colour = 'red';
  }
  if (message) {
    return (
      <div className={`alert-${colour}`}>
        {message}
      </div>
    );
  }

  return null;
};

Alert.propTypes = {
  message: PropTypes.string,
  success: PropTypes.bool,
};

Alert.defaultProps = {
  message: null,
  success: false,
};

export default Alert;
