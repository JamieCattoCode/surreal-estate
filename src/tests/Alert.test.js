import React from 'react';
import { render } from '@testing-library/react';
import Alert from '../components/Alert';

describe('Alert component', () => {
  it('displays an error message for a failed action', () => {
    const { getByText } = render(<Alert message="Error!" />);
    expect(getByText(/error/i).textContent).toBe('Error!');
  });

  it('displays a success message for a successful action', () => {
    const { getByText } = render(<Alert message="Success!!!" success />);
    expect(getByText(/success/i).textContent).toBe('Success!!!');
  });

  it('does not render the component if message is not truthy', () => {
    const { asFragment } = render(<Alert message="" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot if message is error', () => {
    const { getByText, asFragment } = render(<Alert message="Error!" />);
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/error/).textContent).toBe('Error!');
  });

  it('matches snapshot if message is success', () => {
    const { getByText, asFragment } = render(
      <Alert message="Success!" success />,
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Success/).textContent).toBe('Success!');
  });
});
