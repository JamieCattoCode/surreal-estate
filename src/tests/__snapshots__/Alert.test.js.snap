// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component does not render the component if message is not truthy 1`] = `
<DocumentFragment>
  <div>
    Hello!
  </div>
</DocumentFragment>
`;

exports[`Alert component matches snapshot if message is error 1`] = `
<DocumentFragment>
  <div
    class="alert-red"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert component matches snapshot if message is success 1`] = `
<DocumentFragment>
  <div
    class="alert-green"
  >
    Success!
  </div>
</DocumentFragment>
`;
